src/fetchSites.js

```diff
@@ -0,0 +1,82 @@
+const knownInstances = require('./knownInstances.json');
+const fetchP2PLinks = require('./p2pLinks');
+
+const API_ENDPOINT = 'https://api.distributed.press/v1/sites';
+
+async function loadFetch() {
+    const fetchModule = await import('node-fetch');
+    return fetchModule.default || fetchModule;
+}
+
+async function fetchFromAPI() {
+  try {
+    const fetch = await loadFetch();
+    const response = await fetch(API_ENDPOINT);
+    if (!response.ok) {
+      throw new Error(`Failed to fetch API: ${response.statusText}`);
+    }
+
+    const data = await response.json();
+    if (Array.isArray(data)) {
+      return data.map((domain) => ({
+        domain,
+        public: true,
+        links: { http: { enabled: true, link: `https://${domain}` } },
+      }));
+    } else {
+      console.error('API response is not an array of domains:', data);
+      return [];
+    }
+  } catch (error) {
+    console.error(`Error fetching from API: ${error.message}`);
+    return [];
+  }
+}
+
+async function fetchFromKnownInstances() {
+  const sites = [];
+  for (const instance of knownInstances) {
+    try {
+      const fetch = await loadFetch();
+      const response = await fetch(`${instance}/.well-known/distributed.press/sites`);
```

Comment:
Where is this URL coming from, is it necessary?

Reply:
The `.well-known/distributed.press/sites` follows the `.well-known` pattern from [RFC 8615](https://datatracker.ietf.org/doc/html/rfc8615) and aligns with Distributed Press's use of `.well-known`, as seen in [nginx-static.j2](https://github.com/hyphacoop/api.distributed.press/blob/8958235679e35e812a76c467c858869dbc67bfdd/ansible/roles/distributed_press/templates/nginx-static.j2#L23) (e.g., webfinger, host-meta), not explicitly defined there, it extends the .well-known namespace for metadata and discovery, consistent with Distributed Press conventions.

I think this https://github.com/hyphacoop/api.distributed.press/pull/100 will eliminate the need to fetch .well-known endpoints or scrape site metadata. Once merged, I'll update the code to leverage this new API response.

Reply:
Lets skip this feature and stick to the p2p URLs in the site list